queries/places.py

```diff
@@ -0,0 +1,352 @@
+    such as opening hours, location, address, etc. Uses Google's Places API.
+
+"""
+
+
+import logging
+import re
+from datetime import datetime, timedelta
+
+from geo import in_iceland, iceprep_for_street
+from queries import (
+    gen_answer,
+    query_places_api,
+    query_place_details,
+    icequote,
+    numbers_to_neutral,
+)
+from reynir import NounPhrase
+
+
+_PLACES_QTYPE = "Places"
+
+
+# Indicate that this module wants to handle parse trees for queries,
+# as opposed to simple literal text strings
+HANDLE_TREE = True
+
+# The context-free grammar for the queries recognized by this plug-in module
+GRAMMAR = """
+
+Query →
+    QPlacesQuery '?'?
+
+QPlacesQuery →
+    QPlacesOpeningHoursNow | QPlacesIsOpen | QPlacesIsClosed | QPlacesAddress
+
+QPlacesOpeningHoursNow →
+    QPlacesOpeningHours QPlToday?
+
+QPlacesOpeningHours →
+    "hvað" "er" "opið" "lengi" QPlacesPrepAndSubject
+    | "hvað" "er" "lengi" "opið" QPlacesPrepAndSubject
+    | "hverjir" "eru" "opnunartímar" QPlacesPrepAndSubject
+    | "hverjir" "eru" "opnunartímar" QPlacesSubjectEf
+    | "hvaða" "opnunartímar" "eru" QPlacesPrepAndSubject
+    | "hversu" "lengi" "er" "opið" QPlacesPrepAndSubject
+    | "hve" "lengi" "er" "opið" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "opnar" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "opnar" QPlacesSubjectNf
+    | "hvenær" "lokar" QPlacesPrepAndSubject
+    | "hvenær" "lokar" QPlacesSubjectNf
+    | "hvenær" "opnar" QPlacesPrepAndSubject
+    | "hvenær" "opnar" QPlacesSubjectNf
+    | "klukkan" "hvað" "lokar" QPlacesPrepAndSubject
+    | "klukkan" "hvað" "lokar" QPlacesSubjectNf
+    | "hversu" "langt" "er" "í" "lokun" QPlacesPrepAndSubject
+    | "hversu" "langt" "er" "í" "lokun" QPlacesSubjectEf
+    | "hvað" "er" "langt" "í" "lokun" QPlacesPrepAndSubject
+    | "hvað" "er" "langt" "í" "lokun" QPlacesSubjectEf
+    | "hvenær" "er" "opið" QPlacesPrepAndSubject
+    | "hvað" "er" QPlacesSubjectNf QPlOpen "lengi"
+    | "hve" "lengi" "er" QPlacesSubjectNf QPlOpen
+    | "hversu" "lengi" "er" QPlacesSubjectNf QPlOpen
+
+QPlacesIsOpen →
+    "er" "opið" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlOpen QPlNow?
+
+QPlacesIsClosed →
+    "er" "lokað" QPlacesPrepAndSubject QPlNow?
+    | "er" QPlacesSubjectNf QPlClosed QPlNow?
+
+QPlacesAddress →
+    "hvert" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvað" "er" "heimilisfangið" QPlacesPrepAndSubject
+    | "hvert" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvað" "er" "heimilisfang" QPlacesSubjectEf
+    | "hvar" "er" QPlacesSubjectNf "til" "húsa"
+    | "hvar" "er" QPlacesSubjectNf "staðsett" 
+    | "hvar" "er" QPlacesSubjectNf "staðsettur"
+    | QPlacesPreposition "hvaða" "götu" "er" QPlacesSubjectNf
+    | QPlacesPreposition "hvaða" "stræti" "er" QPlacesSubjectNf
+
+QPlacesPrepAndSubject →
+    QPlacesPrepWithÞgf QPlacesSubjectÞgf
+    | QPlacesPrepWithÞf QPlacesSubjectÞf
+
+QPlacesSubjectNf →
```

Comment:
Could be made more succinct with:

```
QPlacesSubject/fall → Nl/fall
$tag(keep) QPlacesSubject/fall
```

... then using `QPlacesSubject_nf` instead of `QPlacesSubjectNf`